### PR TITLE
CCDB-4929: Adding error messages for union operation failure cases.

### PR DESCRIPTION
**Pull request by ManasjyotiSharma**
CCDB-4929: Adding error messages for union operation failure cases.

This issue was seen a bunch of times during on-call and because there was no error message available rather just based on stack trace we couldn't classify these errors previously.

**PS**: Targeting 2.0.x as the original code change got introduced there ([PR-121](https://github.com/confluentinc/kafka-connect-bigquery/pull/121))

### CODE REVIEW

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java [hunk 1]
```diff
@@ -382,8 +382,10 @@ private Field unionizeFields(Field firstField, Field secondField) {
       }
     }
 
-    checkState(firstField.getName().equals(secondField.getName()));
-    checkState(firstField.getType() == secondField.getType());
+    checkState(firstField.getName().equals(secondField.getName()),
+            "Cannot perform union operation on two fields having different names");
+    checkState(firstField.getType() == secondField.getType(),
+            "Cannot perform union operation on two fields having different datatypes");
```
Comment (mukkachaitanya) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java:388]: Uber suggestion @ManasjyotiSharma not inclined either way: Should we add the field name and types at the end of errorMessage? We can base our classification on the static part, but the customers on-prem as well will be able to neatly see what fields are we referring to here.

Comment (ManasjyotiSharma) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java:388]: Good suggestion @mukkachaitanya, let me include that!

File: pom.xml [hunk 2]
```diff
@@ -115,7 +115,7 @@
     <pluginRepositories>
         <pluginRepository>
             <id>confluent</id>
-            <url>http://packages.confluent.io/maven/</url>
+            <url>https://packages.confluent.io/maven/</url>
```
Comment (mukkachaitanya) [inline pom.xml:118]: Can we target these changes in another PR? Looks like this change can be ported as old as 1.1.x. So we can raise a PR to that branch and forward merge it. WDYT?

Comment (ManasjyotiSharma) [reply pom.xml:118]: @mukkachaitanya The PR jenkins build was showing a red cross w.r.t "Code scanning"/CodeQL and so I had made that pom commit to make the build Green. I see your point though. Does the PR merge allow to force/bypass CodeQL check and is that advisable? If so I can revert the pom change from this PR and track that separately against 1.1.x as you pointed.

Comment (mukkachaitanya) [reply pom.xml:118]: > I see your point though. Does the PR merge allow to force/bypass CodeQL check and is that advisable?

I suppose we can. Another way to go about it is to create the PR to 1.1.x first and merge it through and then merge this PR.

Comment (ManasjyotiSharma) [reply pom.xml:118]: @mukkachaitanya I liked the later approach, thanks for the suggestion. Let me first do the pom PR alone and pint merge same, post that merge this current PR.

Comment (ManasjyotiSharma) [reply pom.xml:118]: @mukkachaitanya Raised [PR-229](https://github.com/confluentinc/kafka-connect-bigquery/pull/229). Please approve.

Comment (ManasjyotiSharma) [reply pom.xml]: @mukkachaitanya I have pint merged PR-229. Can you please approve current PR? Thanks!

**Review by mukkachaitanya — APPROVED** [review]
Thanks, @ManasjyotiSharma!

**Pull request merged**